### ISSUE TITLE
TestCase attribute with optional parameters fails with more than one optional parameters

### ISSUE THREAD

**Issue by mauroservienti** [nunit/nunit#1760]
Using `NUnit` 3.4.1 and running the following tests using the `NUnit3-console` runner this one fails

```
[TestCase(1, 10)]
public void should_handle_the_second_optional_arg_when_third_is_missing(int a, int b = 1, bool c = true)
{
    Assert.AreEqual(10, b);
}
```

In this case what happens is that `b` is always 1, the expected behavior is that `b` is 10.
These 2 other work as expected:

```
[TestCase(1, 10, false)]
public void should_handle_all_optional_args(int a, int b = 1, bool c = true)
{
    Assert.AreEqual(10, b);
    Assert.AreEqual(false, c);
}

[TestCase(1, 10)]
public void should_handle_the_second_optional_arg(int a, int b = 1)
{
    Assert.AreEqual(10, b);
}
```

So it seems that the misleading behavior happens when there are more than one optional arguments and  not all of them are specified.

Is this a supported scenario? In case it is not shouldn't it throw?

**Comment by rprouse** [comment]
It is supposed to work. Optional parameters with `TestCase` is a new feature, so I expect that it isn't fully implemented. We likely only tested the one optional parameter case.

See PR #1279, the tests are very limited and do not test your scenario, they only test the passing in or not passing in the default argument which doesn't test if we are overriding the defaults.

@ChrisMaddock do you have time to take a second look at this?

**Comment by ChrisMaddock** [comment]
Sorry, been away.

I think this may be the same issue as #1640, which is fixed waiting for the 3.5 release. Difficult to compare properly on my phone though - @mauroservienti , can you confirm the related pull request fixes your issue? I think one of the new unit tests covers exactly your case, sorry - I goofed a little on the first version.

**Comment by ChrisMaddock** [comment]
Had a look on pc, it looks like #1649 fixes this (which was the fix for #1640). There's also now a unit test for specifically this case.

@mauroservienti - this should be fixed in the 3.5 release. I'll close this issue now, although please reopen if I've missed something!
